test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -1,230 +1,732 @@
+            Set-Location $level2_0
+            $level2Loc = Get-Location
+            $currentLoc.Path | Should Be $root
+            $level1Loc.Path | Should Be $level1_0Full
+            $level2Loc.Path | Should Be $level2_0Full
+        }
+
+        It "Verify Push-Location and Pop-Location" {
+            #push a bunch of locations
+            Push-Location
+            $push0 = Get-Location
+            Set-Location $level1_0
+            Push-Location
+            $push1 = Get-Location
+            Set-Location $level2_0
+            Push-Location
+            $push2 = Get-Location
+
+            #navigate back home to change path out of all pushed locations
+            Set-Location "TestDrive:\"
+
+            #pop locations off
+            Pop-Location
+            $pop0 = Get-Location
+            Pop-Location
+            $pop1 = Get-Location
+            Pop-Location
+            $pop2 = Get-Location
+
+            $pop0.Path | Should Be $push2.Path
+            $pop1.Path | Should Be $push1.Path
+            $pop2.Path | Should Be $push0.Path
+        }
     }
 
-    It "Verify Copy-Item" {
-        $newFile = Copy-Item -Path $testFile -Destination "copyFile.txt" -PassThru
-        $fileExists = Test-Path "copyFile.txt"
-        $fileExists | Should Be $true
-        $newFile.Name | Should Be "copyFile.txt"
+    Context "Validate basic Content Cmdlets for the FileSystem provider" {
+        BeforeEach {
+            New-Item -Path $testFile -ItemType File > $null
+        }
+
+        AfterEach {
+            Remove-Item -Path $testFile -Force -ErrorAction SilentlyContinue
+        }
+
+        It "Set-Content to a file" {
+            $content =  Set-Content -Value "some content" -Path $testFile -PassThru
+            $content | Should BeExactly "some content"
+        } 
+
+        It "Add-Content to a file" {
+            $content = Set-Content -Value "some content" -Path $testFile -PassThru
+            $addContent = Add-Content -Value " more content" -Path $testFile -PassThru
+            $fullContent = Get-Content -Path $testFile
+            $content | Should Match "some content"
+            $addContent | Should Match "more content"
+            ($fullContent[0] + $fullContent[1]) | Should Match "some content more content"
+        }
+
+        It "Clear-Content of a file" {
+            Set-Content -Value "some content" -Path $testFile
+            $contentBefore = Get-Content -Path $testFile
+            Clear-Content -Path $testFile
+            $contentAfter = Get-Content -Path $testFile
+            $contentBefore.Count | Should Be 1
+            $contentAfter.Count | Should Be 0
+        }
     }
 
-    It "Verify Move-Item" {
-        $newFile = Move-Item -Path $testFile -Destination "moveFile.txt" -PassThru
-        $fileExists = Test-Path "moveFile.txt"
-        $fileExists | Should Be $true
-        $newFile.Name | Should Be "moveFile.txt"
+    Context "Validate Resolve-Path Cmdlet Parameters" {
```

Comment:
These should be moved to "Validate Basic Path Cmdlets". Fully testing the cmdlet parameters is a feature-level scenario and this isn't a full test of them